importer/helpers/dagbuilder.go

```diff
@@ -3,6 +3,13 @@ package helpers
 import (
 	dag "github.com/ipfs/go-ipfs/merkledag"
 	"github.com/ipfs/go-ipfs/pin"
+	"sync"
+)
+
+const (
+	hashWorkers  = 1
+	storeWorkers = 50
```

Comment:
I'm willing to bet that if we keep all of these at 1, we will still see significant perf improvements, and we will get slightly cleaner code with not having to worry about multiple different workers.